modules/backup/variables.tf

```diff
@@ -42,6 +42,24 @@ variable "recovery_vault_sku" {
   default     = "Standard"
 }
 
+variable "recovery_vault_storage_mode_type" {
+  description = "The storage type of the Recovery Services Vault. Possible values are `GeoRedundant`, `LocallyRedundant` and `ZoneRedundant`. Defaults to `GeoRedundant`."
+  type        = string
+  default     = "GeoRedundant"
+}
+
+variable "recovery_vault_cross_region_restore_enabled" {
+  description = "Is cross region restore enabled for this Vault? Only can be `true`, when `storage_mode_type` is `GeoRedundant`. Defaults to `false`."
+  type        = bool
+  default     = false
```

Comment:
Maybe it would make sense to have this set to `true` since `recovery_vault_storage_mode_type` default value is `GeoRedundant`. I can't find the specific cost of this feature.

Reply:
I've used the [default value](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/recovery_services_vault#cross_region_restore_enabled) specified in the provider and the API spec.

In the [pricing page](https://azure.microsoft.com/en-us/pricing/details/backup/), they mention the following:

> If you enable cross-region-restore, we **upgrade your backup storage** from GRS to read-access geo-redundant storage (RA-GRS).

Based on which I assume that the price will bump from `€0.0443 per GB (GRS)` to `€0.0563 per GB (RA-GRS)`.

Also, I'm not sure if we should think of the current users who don't want this feature, as they manually need to update their code to opt-out. Considering the number of downloads, I don't think that would be a problem.

What do you think, we use the default or I set it to `true`?

Reply:
After discussing with team, we chose to set it to `true`. For example, paying extra 10€ for 1Tb of data seems fair related to the benefit of this feature.

Reply:
Done!

Reply:
Thanks.

We'll try to find time soon to merge and release your great contributions.